tests/helper/helper_interactive.go

```diff
@@ -0,0 +1,66 @@
+// +build linux darwin dragonfly solaris openbsd netbsd freebsd
+
+package helper
+
+import (
+	"bytes"
+	"log"
+	"os/exec"
+
+	"github.com/Netflix/go-expect"
+	"github.com/hinshun/vt10x"
+	"github.com/kr/pty"
+	. "github.com/onsi/gomega"
+)
+
+func RunInteractive(commonVar CommonVar, command []string, test func(*expect.Console, *bytes.Buffer)) (string, error) {
+
+	ptm, pts, err := pty.Open()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	term := vt10x.New(vt10x.WithWriter(pts))
+
+	c, err := expect.NewConsole(expect.WithStdin(ptm), expect.WithStdout(term), expect.WithCloser(pts, ptm))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer c.Close()
+
+	// execute the command
+	cmd := exec.Command(command[0], command[1:]...)
+	// setup stdin, stdout and stderr
+	cmd.Stdin = c.Tty()
+	cmd.Stdout = c.Tty()
+	cmd.Stderr = c.Tty()
+	err = cmd.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buf := new(bytes.Buffer)
+	test(c, buf)
+	if err != nil {
```

Comment:
But `test` function is not returning anything.

Reply:
Yes, `buf` will contain the output, and we can use it as that is returned by `RunInteractive` method.

Reply:
I meant the error checking here is futile because the `test(c, buf)` call is not returning any error or anything at all, AFAICS.

Reply:
I think we can handle it in the in function like in the example we are checking that err should be nil when using `helper.Sendline` or `helper.Expectline` method. 
Do you have any scenario in mind that require returning error to the `RunInteractive` method?